File: LeetCode/Medium/0050-powx-n/0050-powx-n.js
/**
 * @param {number} x
 * @param {number} n
 * @return {number}
 */
var myPow = function (x, n) {
    if (n === 0) return 1
    let res = 0
    let pow = Math.abs(n)
    if (pow % 2 == 0) res = myPow(x*x,pow/2)
    else res = myPow(x*x,(pow-1)/2) *x
    return n > 0 ? res: (1/res)
};

File: LeetCode/Medium/0050-powx-n/0050-powx-n.py
class Solution:
    def myPow(self, x: float, n: int) -> float:
        if n==0:
            return 1
        pows=abs(n)
        if pows%2==0:
            res=self.myPow(x*x,pows//2)
        else:
            res=self.myPow(x*x,pows//2)*x
        return res if n>0 else (1/res) 
        

        